doc/docker-usage.md

```diff
@@ -0,0 +1,45 @@
+# Build a zcashd node from multi-container Docker build
+
+If no VERSION is exported the Dockerfile will apply the default value `master` and build current master.
+
+```
+#in the root of the zcash source directory:
+export VERSION=v2.0.3
+docker build --build-arg VERSION -t zcash/zcash:$VERSION --memory-swap -1 .
+```
+
+# Create docker volumes and populate the `zcash-params` volume
+```
+docker volume create mainnet-params
+
+docker run --rm -itd \
+--mount source=mainnet-params,destination=/home/zcash/.zcash-params \
+zcash/zcash:latest bash -c "fetch-params.sh"
+```
+
+# Create docker volume to persist the blockchain database and state 
+```
+docker volume create mainnet-chain
+```
+
+# Start a container and background it. Optionally mount in a configuration file or wallet
+```
+CUR_PATH=`PWD`
+
+docker run -itd \
+--name zcash \
+--mount source=mainnet-chain,destination=/home/zcash/.zcash \
+--mount source=mainnet-params,destination=/home/zcash/.zcash-params \
+-v $CUR_PATH/mainnet.daemon.conf:/home/zcash/.zcash/zcash.conf \
```

Comment:
I found this line initially confusing and I think there needs to be more explanation (as this related to the optional part that isn't covered by examples). Perhaps removing it from here then adding a section on how to optionally mount a configuration file would be a better layout for those following along? 

Also, could --mount not be used here also for consistency?

Reply:
One has to use the `-v` to mount in that specific configuration file vs the volume mounts.

Agreed, I'll add more explanation.

Reply:
Can't you do something like `--mount type=bind,source="$(pwd)"/mainnet.daemon.conf,target=/home/zcash/.zcash/zcash.conf` given the file definately exists?

Not that it really matters though it was just lacking a bit of context (provided above). Maybe a section about creating a custom zcash.conf file and then how to mount it in and why you should definately do this if you want to use as a wallet. Then stripping this line out here would ensure anyone blindly following along won't throw an error here.

Reply:
Played around with this which is perhaps the most consistent albeit maybe a little verbose:

```
docker run -itd \
--name zcash
--mount type=volume,source=mainnet-chain,destination=/home/zcash/.zcash \
--mount type=volume,source=mainnet-params,destination=/home/zcash/.zcash-params \
--mount type=bind,source="$(pwd)"/custom-mainnet.conf,target=/home/zcash/.zcash/zcash.conf \
zcash/zcash:latest
```